app/backend/approaches/chatreadretrieveread.py

```diff
@@ -242,3 +264,17 @@ def get_messages_from_history(
 
         messages = message_builder.messages
         return messages
+    
+    def get_search_query(self, chat_completion: any, user_query: str):
+        response_message = chat_completion["choices"][0]["message"]
+        print(response_message)
+        if function_call := response_message.get("function_call"):
+            if function_call["name"] == "search_sources":
+                arg = json.loads(function_call["arguments"])
+                search_query = arg["search_query"]
+                if search_query is not None and search_query != "0":
+                    return search_query
+        elif query_text := response_message.get("content"):
+            if query_text.strip() != "0":
```

Comment:
It might be more clear to make "0" a class constant like NO_RESPONSE, since this "0" is pretty far from the prompt.

Have you actually seen it return 0 in any cases by the way? I don't think I have.

Reply:
It appeared in corner cases where user questions aren't within the scope.